core/src/main/java/org/testcontainers/images/RemoteDockerImage.java

```diff
@@ -42,59 +46,47 @@ protected final String resolve() {
 
         DockerClient dockerClient = DockerClientFactory.instance().client();
         try {
-            int attempts = 0;
-            Exception lastException = null;
-            while (true) {
-                // Does our cache already know the image?
-                if (AVAILABLE_IMAGE_NAME_CACHE.contains(imageName)) {
-                    logger.trace("{} is already in image name cache", imageName);
-                    break;
-                }
-
-                // Update the cache
-                ListImagesCmd listImagesCmd = dockerClient.listImagesCmd();
-
-                if (Boolean.parseBoolean(System.getProperty("useFilter"))) {
-                    listImagesCmd = listImagesCmd.withImageNameFilter(imageName.toString());
-                }
-
-                List<Image> updatedImages = listImagesCmd.exec();
-                updatedImages.stream()
-                    .map(Image::getRepoTags)
-                    .filter(Objects::nonNull)
-                    .flatMap(Stream::of)
-                    .map(DockerImageName::new)
-                    .collect(Collectors.toCollection(() -> AVAILABLE_IMAGE_NAME_CACHE));
-
-                // And now?
-                if (AVAILABLE_IMAGE_NAME_CACHE.contains(imageName)) {
-                    logger.trace("{} is in image name cache following listing of images", imageName);
-                    break;
-                }
-
-                // Log only on first attempt
-                if (attempts == 0) {
-                    logger.info("Pulling docker image: {}. Please be patient; this may take some time but only needs to be done once.", imageName);
-                }
-
-                if (attempts++ >= 3) {
```

Comment:
I see that you removed the retrying logic here. Is it really safe?

Reply:
Yes, this is deliberate. I don't think that we have any more reasons to retry pulls.

A pull timeout was one case where a retry actually used to be useful, in that we could expect huge/slow downloads to maybe fail on the first attempt but succeed after that due to cached layers.

With the new code, we can now tolerate extremely long downloads as long as progress is being made. We shouldn't need to retry downloads because of these any more.

Other reasons for failure, such as unavailable images or auth failures make no sense to retry anyway.

Reply:
I'll leave this comment unresolved for easy visiblity should we ever regret this change!

Reply:
It turns out I was wrong; I've noticed numerous failures in CI due to timeout errors at pull time.

Will reinstate retry logic for pulls!

Reply:
See #1712